### PR TITLE
PlaceInfo use case - Nicolas

### PR DESCRIPTION
**Pull request by nicolasdiasm**
[Nov 27: PlaceInfo use case implemented, back button added to SearchView and Clean Architecture fixes.

New files added: ListingUeCaseFactory, FilePlaceInfoDataAccessObject, PlaceInfoController, PlaceInfoPresenter, PlaceInfoState, PlaceInfoViewModel, PlaceInfoView, PlaceInfoDataAccessInterface, PlaceInfoInteractor, PlaceInfoInputData, PlaceInfoOutputData, PlaceInfoInputBoundary, PlaceInfoOutputBoundary.

Use Case Description: After getting a listing from the search use case, the user can now click on any place and retrieve more information about that specific place (name, address, city, coordinates, tags, and a google maps link).

### CODE REVIEW

**Comment by Arnnav-A** [comment]
We should wait on merging this, after review, until we merge Andrew's PR as the conflicts would be different.

File: src/interface_adapter/listing_results/ListingResultsState.java
Comment (JG808) [inline src/interface_adapter/listing_results/ListingResultsState.java:11]: String place_names is not a Java convention name, it's better to refactor

File: src/interface_adapter/listing_results/ListingResultsState.java
Comment (JG808) [inline src/interface_adapter/listing_results/ListingResultsState.java:33]: add a line between setListing and getPlaceName

File: src/interface_adapter/listing_results/ListingResultsViewModel.java
Comment (JG808) [inline src/interface_adapter/listing_results/ListingResultsViewModel.java:35]: add a line between getPoints and getPlacesName

File: src/interface_adapter/listing_results/ListingResultsViewModel.java
Comment (JG808) [inline src/interface_adapter/listing_results/ListingResultsViewModel.java:51]: add a line between setPoints and setPlacesName

File: src/interface_adapter/place_info/PlaceInfoPresenter.java
Comment (JG808) [inline src/interface_adapter/place_info/PlaceInfoPresenter.java:8]: delete the line

File: src/interface_adapter/place_info/PlaceInfoState.java
Comment (JG808) [inline src/interface_adapter/place_info/PlaceInfoState.java:5]: Refacor place_name to a Java convention

File: src/interface_adapter/place_info/PlaceInfoViewModel.java
Comment (JG808) [inline src/interface_adapter/place_info/PlaceInfoViewModel.java:4]: delete the line

File: src/interface_adapter/listing_results/ListingResultsViewModel.java
Comment (JG808) [inline src/interface_adapter/listing_results/ListingResultsViewModel.java:55]: move it to the front

File: src/use_case/search/SearchInputData.java
Comment (JG808) [inline src/use_case/search/SearchInputData.java:8]: delet the "-"

File: src/use_case/search/SearchOutputData.java
Comment (JG808) [inline src/use_case/search/SearchOutputData.java:10]: delete the line

File: src/use_case/search/SearchOutputData.java
Comment (JG808) [inline src/use_case/search/SearchOutputData.java:9]: Refactor places_names to Java convention

File: src/view/ListingView.java
Comment (JG808) [inline src/view/ListingView.java:7]: delete the line

File: src/view/PlaceInfoView.java
Comment (JG808) [inline src/view/PlaceInfoView.java:6]: delete the line

**Review by JG808 — DISMISSED** [review]
Overall good, just some deletion of lines in import and name convention

**Comment by nicolasdiasm** [comment]
Finished changes

File: src/interface_adapter/place_info/PlaceInfoPresenter.java
Comment (Andrewwf1) [inline src/interface_adapter/place_info/PlaceInfoPresenter.java:26]: How about "place details not found"? This message should not be presented to user via an info window as usual but should pop up a window (for example) to warn user that this place has no information, or just delete this place when grabbing info from api (as we supposed to recommend places to user)

**Review by Andrewwf1 — COMMENTED** [review]
I go through all the modifications and think all the implementations are good. However, when I testing the actual performance, I find the placeInfo displays empty or "place not found" message when the API cannot find this place or lack of information about it. I think this can be improved by adding an if-else statement in placeInfo Presenter to check whether the selected place has useful information to display or not. Or you can choose to delete them during Place Factory, to avoid showing these empty places to user.
image

**Review by Arnnav-A — APPROVED** [review]
Approved by both reviewers, bug found by Andrew has been fixed

**Pull request merged**
